components/header/Header.jsx

```diff
@@ -76,6 +81,29 @@ const Header = () => {
         setAuthData(data);
       }
     };
+  
+     setInterval(() => {
```

Comment:
This should be cleaned up in the `return` statement of `useEffect` by using [`clearInterval`](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval). Otherwise, every time this component is removed from the DOM and recreated, another one of these timers will be created, which would cause more and more requests to queue up more quickly when the `token` is present.